packages/liveblocks-node/src/thread-notification-helpers.ts

```diff
@@ -0,0 +1,284 @@
+import type {
+  BaseUserMeta,
+  CommentBody,
+  CommentBodyJson,
+  CommentData,
+  DRI,
+  DU,
+  OptionalPromise,
+  ResolveUsersArgs,
+} from "@liveblocks/core";
+import {
+  generateCommentUrl,
+  getMentionedIdsFromCommentBody,
+  transformCommentBody,
+} from "@liveblocks/core";
+
+import type { Liveblocks } from "./client";
+import type { ThreadNotificationEvent } from "./webhooks";
+
+export type ThreadNotificationCommentData = Omit<
+  CommentData,
+  "body" | "deletedAt"
+> & {
+  body: CommentBody;
+  deletedAt?: never;
+};
+
+export type ThreadNotificationUnreadRepliesData = {
+  type: "unreadReplies";
+  comments: ThreadNotificationCommentData[];
+};
+
+export type ThreadNotificationUnreadMentionData = {
+  type: "unreadMention";
+  comments: ThreadNotificationCommentData[];
+};
+
+export type ThreadNotificationData =
+  | ThreadNotificationUnreadRepliesData
+  | ThreadNotificationUnreadMentionData;
+
+/** @internal */
+export const getLastCommentWithMention = ({
+  comments,
+  mentionedUserId,
+}: {
+  comments: ThreadNotificationCommentData[];
+  mentionedUserId: string;
+}): ThreadNotificationCommentData | null => {
+  return (
+    Array.from(comments)
+      .reverse()
+      .filter((c) => c.userId !== mentionedUserId)
+      .find((c) => {
+        const mentionedUserIds = getMentionedIdsFromCommentBody(c.body);
+        return mentionedUserIds.includes(mentionedUserId);
+      }) ?? null
+  );
+};
+
+/**
+ * Get thread notification data helper.
+ *
+ * It returns either an object containing a list of unread replies for a thread
+ * or either an object containing a list of the last unread comment where the notification
+ * receiver was mentioned in.
+ *
+ * @param params.client Liveblocks node client
+ * @param params.event The thread notification event
+ * @returns A thread notification data object
+ *
+ * @example Unread replies:
+ * {
+ *  type: "unreadReplies",
+ *  comments: [unread_comment_0, unread_comment_1, ...],
+ * }
+ *
+ * @example Unread mention:
+ * {
+ *  type: "unreadMention",
+ *  comments: [unread_comment_with_mention]
+ * }
+ */
+export async function getThreadNotificationData(params: {
+  client: Liveblocks;
+  event: ThreadNotificationEvent;
+}): Promise<ThreadNotificationData> {
+  const { client, event } = params;
+  const { threadId, roomId, userId, inboxNotificationId } = event.data;
+
+  const [thread, inboxNotification] = await Promise.all([
+    client.getThread({ roomId, threadId }),
+    client.getInboxNotification({ inboxNotificationId, userId }),
+  ]);
+
+  const readAt = inboxNotification.readAt;
+  const unreadComments = thread.comments
+    .filter((c) => c.userId !== userId)
+    .filter((c) => c.body !== undefined && c.deletedAt === undefined)
+    .filter((c) =>
+      readAt
+        ? c.createdAt > readAt && c.createdAt <= inboxNotification.notifiedAt
+        : c.createdAt <= inboxNotification.notifiedAt
+    ) as ThreadNotificationCommentData[];
```

Comment:
Maybe this logic could be extracted like `getLastCommentWithMention` and also kept internal, just for clarity

Reply:
Yep sounds good 👍🏻